exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/CreateFunctionHandler.java

```diff
@@ -175,22 +175,20 @@ private void initRemoteRegistration(List<String> functions,
     List<Jar> remoteJars = remoteRegistry.getRegistry(version).getJarList();
     validateAgainstRemoteRegistry(remoteJars, jarManager.getBinaryName(), functions);
     jarManager.copyToRegistryArea();
-    boolean cleanUp = true;
     List<Jar> jars = Lists.newArrayList(remoteJars);
     jars.add(Jar.newBuilder().setName(jarManager.getBinaryName()).addAllFunctionSignature(functions).build());
     Registry updatedRegistry = Registry.newBuilder().addAllJar(jars).build();
     try {
       remoteRegistry.updateRegistry(updatedRegistry, version);
-      cleanUp = false;
     } catch (VersionMismatchException ex) {
+      jarManager.deleteQuietlyFromRegistryArea();
```

Comment:
I wonder about the retry approach. If we get a version mis-match, it means that something has changed: some other agent is modifying the registry at the same time. How should this be handed? Fixed number of retry? Wait and retry? Presumably, after some time, the system will get less busy, or we'll finally get our crack at the registry.

Also, is it wise to use recursion to retry? Doing so causes a stack depth proportional to the retry count, which may not be ideal. Loop instead?